src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsUrlTest.java

```diff
@@ -0,0 +1,186 @@
+
+package com.puppycrawl.tools.checkstyle.internal;
+
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.parsers.SAXParser;
+import javax.xml.parsers.SAXParserFactory;
+
+import org.junit.jupiter.api.Test;
+import org.xml.sax.Attributes;
+import org.xml.sax.helpers.DefaultHandler;
+
+public class XdocsUrlTest {
+
+    public static final String PACKAGE_NAME =
+            "src/main/java/com/puppycrawl/tools/checkstyle/checks";
+
+    public static final String NODE_NAME = "a";
+
+    public static final String SUFFIX_JAVA = ".java";
+
+    public static final String SUFFIX_CHECK_JAVA = "Check.java";
+
+    public static final String MISC = "misc";
+
+    public static final String ANNOTATION = "annotation";
+
+    public static final String COMMENTS_INDENTATION = "CommentsIndentation";
+
+    public static final String INDENTATION = "Indentation";
+
+    public static final String SUPPRESS_WARNINGS_HOLDER = "SuppressWarningsHolder";
+
+    public static final Path AVAILABLE_CHECKS_PATH = Paths.get("src/xdocs/checks.xml");
+
+    private static Map<String, String[]> getXdocsMap() {
+        final File checks = new File(PACKAGE_NAME);
+        final String[] checksSubDirNames = checks.list((rootDir, pathname) -> {
+            return !pathname.endsWith(SUFFIX_JAVA);
+        });
+        final Map<String, String[]> checksNamesMap = new HashMap<>();
+        if (checksSubDirNames != null) {
+            final FilenameFilter filter = (rootDir, pathname) -> {
+                return pathname.endsWith(SUFFIX_CHECK_JAVA);
+            };
+            for (String dirName : checksSubDirNames) {
+                final File dirFile = new File(PACKAGE_NAME + "/" + dirName);
+                final String[] fileNames = dirFile.list(filter);
+                checksNamesMap.put(dirName, fileNames);
+            }
+            final String[] miscNames = checks.list(filter);
+            checksNamesMap.put(MISC, miscNames);
+        }
+        return checksNamesMap;
+    }
+
+    @Test
+    public void testXdocsUrl() throws Exception {
+        final SAXParserFactory parserFactory = SAXParserFactory.newInstance();
+        final SAXParser parser = parserFactory.newSAXParser();
+        final CheckTest checkHandler = new CheckTest(NODE_NAME);
+        try (InputStream input = Files.newInputStream(AVAILABLE_CHECKS_PATH)) {
+            parser.parse(input, checkHandler);
+        }
+        final Map<String, String[]> checksNamesMap = getXdocsMap();
+        for (List<String> sub : checkHandler.list) {
+            final String moduleName = sub.get(0);
+            final String checkName1 = sub.get(1);
+            final String checkName2 = sub.get(2);
```

Comment:
`checkName1` and `checkName2` please rename to make it more understandable

Reply:
done